data-resource/Cargo.toml

```diff
@@ -8,23 +8,35 @@ name = "data_resource"
 crate-type = ["rlib"]
 bench = false
 
+[[bench]]
+name = "bench_resource_id_creation"
+harness = false
+path = "benches/id_creation_benchmarks.rs"
+
 [dependencies]
 data-error = { path = "../data-error" }
-fs-atomic-versions = { path = "../fs-atomic-versions" }
-
-log = { version = "0.4.17", features = ["release_max_level_off"] }
-serde = { version = "1.0.138", features = ["derive"] }
-crc32fast = "1.3.2"
-anyhow = "1"
+serde = { version = "1.0", features = ["derive"] }
+log = "0.4"
 
+# Hash type specefic dependencies
+# CRC32
+crc32fast = { version = "1.3", optional = true }
+# Blake3
+blake3 = { version = "1.5", optional = true }
+hex = { version = "0.4", optional = true }
 
 [dev-dependencies]
-# benchmarking
+# Benchmarks
 criterion = { version = "0.5", features = ["html_reports"] }
-pprof = { version = "0.13", features = ["criterion", "flamegraph"] }
 rand = "0.8"
+anyhow = "1.0"
 
-[[bench]]
-name = "compute_bytes_benchmark"
-harness = false
-path = "benches/compute_bytes_benchmark.rs"
+
+# Features:
+# - `non-cryptographic-hash`: Uses a non-cryptographic hash functions to define `ResourceId`
+# - `cryptographic-hash`: Uses a cryptographic hash functions to define `ResourceId`
+[features]
+non-cryptographic-hash = ["crc32fast"]
+cryptographic-hash = ["blake3", "hex"]
```

Comment:
What does `"hex"` mean here?

Reply:
It's the [hex](https://crates.io/crates/hex) crate
We currently use it in `cryptographic-hash` feature because we are hex encoding the `ResourceId` and storing it as a string

Reply:
I made some changes to the dependency structure to be more clear. Let me know your thoughts on it.